docs/developer-docs/latest/development/custom-fields.md

```diff
@@ -0,0 +1,340 @@
+
+| Options parameter | Description                                                                     | Type                    |
+| -------------- | ------------------------------------------------------------------------------- | ----------------------- |
+| `base`         | Settings available in the _Base settings_ tab of the field in the Content-type Builder       | `Object` or  `Array of Objects` |
+| `advanced`     | Settings available in the _Advanced settings_ tab of the field in the Content-type Builder   | `Object` or  `Array of Objects` |
+| `validator`    | Validator function returning an object, used to sanitize input | `Function`              |
+
+Both `base` and `advanced` settings accept an object or an array of objects, each object being a settings section. Each settings section must include:
+
+- a `sectionTitle` to declare the title of the section as an [`IntlObject`](https://formatjs.io/docs/react-intl/)
+- and a list of `items` as an array of objects.
+
+Each object in the `items` array can contain the following parameters:
+
+<!-- TODO: fix the table content as the parameters given in the [example from the RFC](https://github.com/strapi/rfcs/blob/3eb034e169746558315d719ca5fb49cec854640a/rfcs/xxxx-custom-fields-api.md#example) and the [TS interface](https://github.com/strapi/rfcs/blob/3eb034e169746558315d719ca5fb49cec854640a/rfcs/xxxx-custom-fields-api.md#admin) described in the RFC are inconsistent 🤷  -->
+
+| Items parameter | Description                                                                                  | Type     |
+| --------------- | -------------------------------------------------------------------------------------------- | -------- |
+| `name`          | Name of the setting item<br/>Must use the `options.setting-name` format.                     | `String` |
+| `description`   | Description of the setting item to use in the Content-type Builder                           | `String` |
+| `intlLabel`     | Translation for the abel of the setting item                                                 | [`IntlObject`](https://formatjs.io/docs/react-intl/) |
+| `type`          | Type of setting (see [Setting types and specific options](#))                                 | `String` |
+
+<!-- TODO: add a super simple example with one base setting, one advanced setting, and a simple validator -->
+::: details Example: Declaring options for an example "color" custom field:
+
+```jsx
+// path: ./src/plugins/my-custom-field-plugin/strapi-admin.js
+
+register(app) {
+  app.customFields.register({
+    // …
+    options: {
+      base: [
+        {
+          sectionTitle: {
+            id: 'color-picker.color.section.format',
+            defaultMessage: 'Format',
+          },
+          items: [
+            {
+              intlLabel: {
+                id: 'color-picker.color.format.label',
+                defaultMessage: 'Color format',
+              },
+              name: 'options.format',
```

Comment:
I guess this was sort of a typo on the RFC. This is the path for the option so that means this option will be found on the `schema.json` as `fieldName.options.format`, that means the validator below needs to be updated to match this path. `fieldName.options['color-picker'].format`, I don't know if that's clear so just ping me to talk about it :)